nemo/collections/tts/models/waveglow.py

```diff
@@ -0,0 +1,231 @@
+            'log': {'loss': loss},
+        }
+        return output
+
+    def validation_step(self, batch, batch_idx):
+        self.mode = OperationMode.validation
+        self.waveglow.mode = OperationMode.validation
+        audio, audio_len = batch
+        z, log_s_list, log_det_W_list, audio_pred, spec, spec_len = self.forward(
+            audio=audio, audio_len=audio_len, run_inverse=(batch_idx == 0)
+        )
+        loss = self.loss(z=z, log_s_list=log_s_list, log_det_W_list=log_det_W_list, sigma=self.sigma)
+        return {
+            "val_loss": loss,
+            "audio_pred": audio_pred,
+            "mel_target": spec,
+            "mel_len": spec_len,
+        }
+
+    def validation_epoch_end(self, outputs):
+        waveglow_log_to_tb_func(
+            self.logger.experiment,
+            outputs[0].values(),
+            self.global_step,
+            tag="eval",
+            mel_fb=self.audio_to_melspec_precessor.fb,
+        )
+        avg_loss = torch.stack([x['val_loss'] for x in outputs]).mean()
+        tensorboard_logs = {'val_loss': avg_loss}
+        return {'val_loss': avg_loss, 'log': tensorboard_logs}
+
+    def __setup_dataloader_from_config(self, cfg, shuffle_should_be: bool = True, name: str = "train"):
+        if "dataset" not in cfg or not isinstance(cfg.dataset, DictConfig):
+            raise ValueError(f"No dataset for {name}")  # TODO
+        if "dataloader_params" not in cfg or not isinstance(cfg.dataloader_params, DictConfig):
+            raise ValueError(f"No dataloder_params for {name}")  # TODO
+        if shuffle_should_be:
+            if 'shuffle' not in cfg.dataloader_params:
+                logging.warning(
+                    f"Shuffle should be set to True for {self}'s {name} dataloader but was not found in its "
+                    "config. Manually setting to True"
+                )
+                with open_dict(cfg["dataloader_params"]):
+                    cfg.dataloader_params.shuffle = True
+            elif not cfg.dataloader_params.shuffle:
+                logging.error(f"The {name} dataloader for {self} has shuffle set to False!!!")
+        elif not shuffle_should_be and cfg.dataloader_params.shuffle:
+            logging.error(f"The {name} dataloader for {self} has shuffle set to True!!!")
+
+        dataset = WaveGlowModel.from_config_dict(cfg.dataset)
+        return torch.utils.data.DataLoader(dataset, collate_fn=dataset.collate_fn, **cfg.dataloader_params)
+
+    def setup_training_data(self, cfg):
+        self._train_dl = self.__setup_dataloader_from_config(cfg)
+
+    def setup_validation_data(self, cfg):
+        self._validation_dl = self.__setup_dataloader_from_config(cfg, shuffle_should_be=False, name="validation")
+
+    @classmethod
+    def list_available_models(cls) -> 'Optional[Dict[str, str]]':
+        """TODO: Implement me!"""
+        pass
+
+    @classmethod
+    def from_pretrained(cls, name: str):
```

Comment:
Please remove this method (you can do so if you rebase/merge latest candidate into your branch)

Reply:
Done in next commit.